### ISSUE TITLE
Error at `&me.time_created`

### ISSUE THREAD

**Issue by hajsf** [jgallagher/rusqlite#438]
I'm trying to use the example given in the [crates.io](https://crates.io/crates/rusqlite):

```rust
extern crate rusqlite;
extern crate time;

use time::Timespec;
use rusqlite::Connection;

#[derive(Debug)]
struct Person {
    id: i32,
    name: String,
    time_created: Timespec,
    data: Option<Vec<u8>>
}

fn main() {
    let conn = Connection::open_in_memory().unwrap();

    conn.execute("CREATE TABLE person (
                  id              INTEGER PRIMARY KEY,
                  name            TEXT NOT NULL,
                  time_created    TEXT NOT NULL,
                  data            BLOB
                  )", &[]).unwrap();
    let me = Person {
        id: 0,
        name: "Steven".to_string(),
        time_created: time::get_time(),
        data: None
    };
    conn.execute("INSERT INTO person (name, time_created, data)
                  VALUES (?1, ?2, ?3)",
                 &[&me.name, &me.time_created, &me.data]).unwrap();

    let mut stmt = conn.prepare("SELECT id, name, time_created, data FROM person").unwrap();
    let person_iter = stmt.query_map(&[], |row| {
        Person {
            id: row.get(0),
            name: row.get(1),
            time_created: row.get(2),
            data: row.get(3)
        }
    }).unwrap();

    for person in person_iter {
        println!("Found person {:?}", person.unwrap());
    }
}
```
But I got the below error:

```
error[E0308]: mismatched types
  --> src/main.rs:32:30
   |
32 |                  &[&me.name, &me.time_created, &me.data]).unwrap();
   |                              ^^^^^^^^^^^^^^^^ expected struct `std::string::String`, found struct `time::Timespec`
   |
   = note: expected type `&std::string::String`
              found type `&time::Timespec`
```

![err](https://user-images.githubusercontent.com/26045838/49389911-8536f200-f738-11e8-9a39-3771bda7e2b0.png)

**Comment by gwenn** [comment]
Sorry, the example on crates.io is wrong.
See https://github.com/jgallagher/rusqlite/pull/417/files for a fixed version.
Or https://github.com/jgallagher/rusqlite/blob/master/README.md.

**Comment by gwenn** [comment]
Also see https://github.com/jgallagher/rusqlite/issues/424#issuecomment-436754481 for an explanation.

**Comment by hajsf** [comment]
Thanks @gwenn I recommend to fix it it the `crates.io`, today I made the same mistake by copying the code there :)

**Comment by gwenn** [comment]
Fixed by 0.16.0 release.
